newrelic/resource_newrelic_synthetics_monitor.go

```diff
@@ -95,158 +92,402 @@ func resourceNewRelicSyntheticsMonitor() *schema.Resource {
 				Optional:    true,
 				Description: "Fail the monitor check if redirected.",
 			},
+			"runtime_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The runtime type that the monitor will run",
+			},
+			"runtime_type_version": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The specific version of the runtime type selected",
+			},
+			"script_language": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The programing language that should execute the script",
+			},
+			"tag": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				MinItems:    1,
+				Description: "The tags that will be associated with the monitor",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Name of the tag key",
+						},
+						"values": {
+							Type:        schema.TypeList,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Required:    true,
+							Description: "Values associated with the tag key",
+						},
+					},
+				},
+			},
+			"enable_screenshot_on_failure_and_script": {
+				Type:        schema.TypeBool,
+				Description: "Capture a screenshot during job execution",
+				Optional:    true,
+			},
+			"custom_headers": {
+				Type:        schema.TypeSet,
+				Description: "Custom headers to use in monitor job",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "Header name",
+							Optional:    true,
+						},
+						"value": {
+							Type:        schema.TypeString,
+							Description: "Header value",
+							Optional:    true,
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
-func buildSyntheticsMonitorStruct(d *schema.ResourceData) synthetics.Monitor {
-	monitor := synthetics.Monitor{
-		Name:         d.Get("name").(string),
-		Type:         synthetics.MonitorType(d.Get("type").(string)),
-		Frequency:    uint(d.Get("frequency").(int)),
-		Status:       synthetics.MonitorStatusType(d.Get("status").(string)),
-		SLAThreshold: d.Get("sla_threshold").(float64),
+//func to build the input to create simple browser monitor
+func buildSyntheticsSimpleBrowserMonitor(d *schema.ResourceData) synthetics.SyntheticsCreateSimpleBrowserMonitorInput {
```

Comment:
Can we move these helpers to the structures file? That way both types of monitor resources are the same.

Reply:
Moved helper files to structure file.